server/worker/jira.go

```diff
@@ -1,31 +1,142 @@
 package worker
 
 import (
+	"bytes"
 	"context"
 	"encoding/json"
+	"fmt"
+	"html/template"
 
+	jira "github.com/andygrunwald/go-jira"
+	"github.com/fleetdm/fleet/v4/server/contexts/ctxerr"
 	"github.com/fleetdm/fleet/v4/server/fleet"
 	kitlog "github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
 )
 
-type Jira struct {
-	ds  fleet.Datastore
-	log kitlog.Logger
-	// TODO: add jira client
+const (
+	// JiraName is the name of the job as registered in the worker.
+	JiraName = "jira"
+
+	nvdCVEURL = "https://nvd.nist.gov/vuln/detail/"
+)
+
+var jiraSummaryTmpl = template.Must(template.New("").Parse(
+	`{{ .CVE }} detected on hosts`,
+))
+
+var jiraDescriptionTmpl = template.Must(template.New("").Parse(
+	`See vulnerability (CVE) details in National Vulnerability Database (NVD) here: {{ .NVDURL }}
+
+See all hosts affected by this vulnerability (CVE) in Fleet: {{ .FleetURL }}
+
+--
+
+This issue was created automatically by your Fleet to Jira integration.
+`,
+))
+
+type jiraTemplateArgs struct {
+	CVE      string
+	NVDURL   string
+	FleetURL string
 }
 
-func NewJira(ds fleet.Datastore, log kitlog.Logger) *Jira {
-	return &Jira{
-		ds:  ds,
-		log: log,
-	}
+// JiraClient defines the method required for the client that makes API calls
+// to Jira.
+type JiraClient interface {
+	CreateIssue(ctx context.Context, issue *jira.Issue) (*jira.Issue, error)
+}
+
+// Jira is the job processor for jira integrations.
+type Jira struct {
+	FleetURL   string
+	Datastore  fleet.Datastore // TODO: we may not need the datastore, though it depends on the URL issue
+	Log        kitlog.Logger
+	JiraClient JiraClient
 }
 
+// Name returns the name of the job.
 func (j *Jira) Name() string {
-	return "jira"
+	return JiraName
 }
 
+// JiraArgs are the arguments for the Jira integration job.
+type JiraArgs struct {
+	CVE  string   `json:"cve"`
+	CPEs []string `json:"cpes"`
+}
+
+// Run processes a worker message for the Jira integration.
 func (j *Jira) Run(ctx context.Context, argsJSON json.RawMessage) error {
```

Comment:
nit: Maybe `Run` should be renamed to something like `ProcessMessage`?

Reply:
Do you have a particular reason why `Run` is bad?

Reply:
It makes sense to me - the worker processes jobs, the jobs' implementations "run" themselves.

Reply:
Nbd then :), just a nit.